uaaextras/clients/uaa.py

```diff
@@ -466,16 +466,15 @@ def invalidate_tokens(self, user_id, zone_id=None, zone_subdomain=None) -> None:
         )
 
 
-    def check_token_valid(self, maybe_valid_token, client_id, client_secret) -> bool:
+    def check_token_valid(self, maybe_valid_token) -> bool:
         """
         Check whether the provided token is valid, using the client_id and client_secret to authenticate
         """
-        our_token = self._get_client_token(client_id, client_secret)
         headers = {
-            "Authorization": f"bearer {our_token}",
+            "Authorization": f"bearer {maybe_valid_token}",
         }
         url = urljoin(self.base_url.rstrip("/"), "/introspect".lstrip("/"))
         response = requests.post(url, data={"token": maybe_valid_token}, headers=headers)
         if response.status_code != 200:
```

Comment:
since we're using a token that might be bad to check if our token is good, a lot of error classes might indicate our token is bad.
It would probably be good to narrow this down to prevent users from looping in case there's a case I'm missing, but I think this is safe enough for now